drawBot/context/baseContext.py

```diff
@@ -403,14 +412,30 @@ def _getCGPath(self):
             elif instruction == AppKit.NSClosePathBezierPathElement:
                 Quartz.CGPathCloseSubpath(path)
         # hacking to get a proper close path at the end of the path
-        x, y, _, _ = self.bounds()
-        Quartz.CGPathMoveToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathAddLineToPoint(path, None, x, y)
-        Quartz.CGPathCloseSubpath(path)
+        if withBounds:
+            x, y, _, _ = self.bounds()
+            Quartz.CGPathMoveToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathAddLineToPoint(path, None, x, y)
+            Quartz.CGPathCloseSubpath(path)
         return path
 
+    def _setCGpath(self, cgpath):
+        self._path = AppKit.NSBezierPath.alloc().init()
+
+        def _addPoints(arg, element):
+            instruction, points = element.type, element.points
+            if instruction == Quartz.kCGPathElementMoveToPoint:
+                self._path.moveToPoint_((points[0].x, points[0].y))
```

Comment:
the points are already valid nsBezierPath points
you can provide the point directly: `moveToPoint_(points[0])` same for `lineToPoint_` and `curveToPoint_controlPoint1_controlPoint2_`

Reply:
Thanks!